src/main/java/br/com/deveficiente/calendario/model/Evento.java

```diff
@@ -0,0 +1,63 @@
+package br.com.deveficiente.calendario.model;
+
+import org.hibernate.validator.constraints.Length;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToMany;
+import javax.persistence.ManyToOne;
+import javax.validation.constraints.Future;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Entity
+public class Evento {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotBlank
+    @Length(max = 100)
+    private final String titulo;
+
+    @Future
+    @NotNull
+    private final LocalDateTime inico;
+
+    @Future
+    @NotNull
+    private final LocalDateTime termino;
+
+    @NotBlank
+    @Length(max = 255)
+    private final String descricao;
+
+    @ManyToMany
+    private final List<Usuario> convidados;
+
+    @NotNull
+    @ManyToOne
+    private final Agenda agenda;
+
+    @NotNull
+    @ManyToOne
+    private final Usuario dono;
+
+    public Evento(@NotBlank @Length(max = 100) final String titulo, @Future @NotNull final LocalDateTime inico,
+                  @Future @NotNull final LocalDateTime termino, @NotBlank @Length(max = 255) final String descricao,
+                  final List<Usuario> convidados, @NotNull final Agenda agenda,
```

Comment:
Aqui me surgiu uma dúvida que eu tinha deixado passar no formulário :). No cartão eu só disse que os emails precisavam ser de usuários que existem. Por outro lado não falei que precisava ter pelo menos um convidado. 

No construtor você obriga a passagem dos convidados, mas no formulário não obriga ter pelo menos um email. Já que você obrigou aqui(e tá tudo bem), vale colocar um @Size lá na lista de emails do form e daqui da entidade.

Reply:
A minha ideia era permitir que a lista de convidados fosse passada vazia ou nula no form, não acho que faça sentido forçar a ter convidados no evento.

Inicializei a lista no form e o set no evento automaticamente pra tentar resolver isso. Testei aqui com lista vazia e nula (omitida, na verdade) e parece que deu certo.

O único ponto que pode dar ruim é se a lista no `NovoEventoForm` for explicitamente setada como nula, dai quebra o construtor do `Evento` que agora tá usando `addAll`. Ao não passar a lista chamando o método POST, parece que o setter dela não foi chamado, mas ainda assim existiria essa possibilidade.

Como alternativa eu vejo:

- Não usar um construtor para criar o `Evento`
- Verificar se a lista passada para o construtor é nula antes de chamar o `addAll`

Confesso não que não gosto muito de nenhuma das duas. Talvez o melhor seja pelo menos pedir que a lista não seja nula no form. O que você acha?

Reply:
O validador da lista de e-mails tb quebra se ela for explicitamente nula 😞 

Me parece que faz sentido deixar ela como `@NotNull` mesmo